File: JS/01.Basics/99.Pre-Exam/01.CatDiet.js
function calculateCalories(inputArray) {
    const [fatPercentage, proteinPercentage, carbohydratePercentage, totalCalories, waterPercentage] = inputArray;

    const fatGrams = (fatPercentage / 100) * totalCalories / 9;
    const proteinGrams = (proteinPercentage / 100) * totalCalories / 4;
    const carbohydrateGrams = (carbohydratePercentage / 100) * totalCalories / 4;

    const totalWeight = fatGrams + proteinGrams + carbohydrateGrams;

    const waterCalories = (waterPercentage / 100) * totalCalories;
    const caloriesPerGram = totalCalories / totalWeight;
    const caloriesPerGramWithoutWater = caloriesPerGram - (waterCalories / totalWeight);

    console.log(caloriesPerGramWithoutWater.toFixed(4));
}

const inputArray1 = [60, 25, 15, 2500, 60];
const inputArray2 = [40, 40, 20, 3000, 40];
const inputArray3 = [20, 60, 20, 1800, 50];

calculateCalories(inputArray1);
calculateCalories(inputArray2);
calculateCalories(inputArray3);

File: Python/01.ProgrammingBasics/99.Pre-Exam/01.CatDiet/_01.CatDiet.py

fat_percentage = int(input())
protein_percentage = int(input())
carbohydrate_percentage = int(input())
total_calories = int(input())
water_percentage = int(input())

fat_grams = (fat_percentage / 100) * total_calories / 9
protein_grams = (protein_percentage / 100) * total_calories / 4
carbohydrate_grams = (carbohydrate_percentage / 100) * total_calories / 4

total_weight = fat_grams + protein_grams + carbohydrate_grams

water_calories = (water_percentage / 100) * total_calories
calories_per_gram = total_calories / total_weight
calories_per_gram_without_water = calories_per_gram - (water_calories / total_weight)

print(f"{calories_per_gram_without_water:.4f}")
